cycode/cli/commands/configure/configure_command.py

```diff
@@ -0,0 +1,137 @@
+from typing import Optional
+
+import click
+
+from cycode.cli import config, consts
+from cycode.cli.user_settings.configuration_manager import ConfigurationManager
+from cycode.cli.user_settings.credentials_manager import CredentialsManager
+from cycode.cli.utils.string_utils import obfuscate_text
+
+_URLS_UPDATED_SUCCESSFULLY_MESSAGE = 'Successfully configured Cycode URLs! Saved to: {filename}'
+_URLS_ARE_SET_IN_ENVIRONMENT_VARIABLES_MESSAGE = (
+    'Note that the URLs (APP and API) that already exist in environment variables '
+    f'({consts.CYCODE_API_URL_ENV_VAR_NAME} and {consts.CYCODE_APP_URL_ENV_VAR_NAME}) '
+    'take precedent over these URLs; either update or remove the environment variables.'
+)
+_CREDENTIALS_UPDATED_SUCCESSFULLY_MESSAGE = 'Successfully configured CLI credentials! Saved to: {filename}'
+_CREDENTIALS_ARE_SET_IN_ENVIRONMENT_VARIABLES_MESSAGE = (
+    'Note that the credentials that already exist in environment variables '
+    f'({config.CYCODE_CLIENT_ID_ENV_VAR_NAME} and {config.CYCODE_CLIENT_SECRET_ENV_VAR_NAME}) '
+    'take precedent over these credentials; either update or remove the environment variables.'
+)
+_CREDENTIALS_MANAGER = CredentialsManager()
+_CONFIGURATION_MANAGER = ConfigurationManager()
+
+
+@click.command(short_help='Initial command to configure your CLI client authentication.')
+def configure_command() -> None:
+    """Configure your CLI client authentication manually."""
+    global_config_manager = _CONFIGURATION_MANAGER.global_config_file_manager
+
+    current_api_url = global_config_manager.get_api_url()
+    current_app_url = global_config_manager.get_app_url()
+    api_url = _get_api_url_input(current_api_url)
+    app_url = _get_app_url_input(current_app_url)
+
+    config_updated = False
+    if _should_update_value(current_api_url, api_url):
+        global_config_manager.update_api_base_url(api_url)
+        config_updated = True
+    if _should_update_value(current_app_url, app_url):
+        global_config_manager.update_app_base_url(app_url)
+        config_updated = True
+
+    current_client_id, current_client_secret = _CREDENTIALS_MANAGER.get_credentials_from_file()
+    client_id = _get_client_id_input(current_client_id)
+    client_secret = _get_client_secret_input(current_client_secret)
+
+    credentials_updated = False
+    if _should_update_value(current_client_id, client_id) or _should_update_value(current_client_secret, client_secret):
+        credentials_updated = True
+        _CREDENTIALS_MANAGER.update_credentials_file(client_id, client_secret)
+
+    if config_updated:
+        click.echo(_get_urls_update_result_message())
+    if credentials_updated:
+        click.echo(_get_credentials_update_result_message())
+
+
+def _get_client_id_input(current_client_id: Optional[str]) -> Optional[str]:
+    prompt_text = 'Cycode Client ID'
+
+    prompt_suffix = ' (optional): '
```

Comment:
The previous behavior was only about the client id and client secret. So the whole "configure" command was only about credentials. I extended this command and want to provide the ability to setup URLs without credentials. To be able to use "cycode auth" command later with on premise installation. Otherwise. If I delete "optional" we push our users to login manually instead of "cycode auth" (with browser)

Reply:
The another way is to create subcommands for "cycode configure"

For example "cycode configure auth", "cycode configure url"

Reply:
Lets stick with one command. Maybe just '[]' in case user did not configure credentials yet. wdyt?

Reply:
I tried this. But when I saw empty brackets I changed to optional ASAP because it looked strange 😅 and not clear from UX (imho)

Reply:
we replicated the same behaviour of `aws configure`

Reply:
done 

<img width="807" alt="image" src="https://github.com/cycodehq-public/cycode-cli/assets/15520314/200e8304-f8d2-4e53-9329-d1d7c7beaab4">